### PR TITLE
Github intergration

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

File: modules/build/controllers/build/controller.go [hunk 1]
```diff
@@ -50,6 +50,24 @@ func (h handler) updateService(key string, build *tektonv1alpha1.TaskRun) (*tekt
...
+	} else if !build.IsDone() {
+		state = "in_progress"
+	}
+	if build.Labels["gitcommit-name"] != "" {
+		gitcommit, err := h.gitcommits.Cache().Get(build.Namespace, build.Labels["gitcommit-name"])
+		if err != nil {
+			return build, err
+		}
+		gitcommit = gitcommit.DeepCopy()
+		gitcommit.Status.BuildStatus = state
```
Comment (ibuildthecloud) [inline modules/build/controllers/build/controller.go]: only update if state really changed

File: modules/build/controllers/gitcommit/controller.go [hunk 2]
```diff
@@ -51,6 +61,55 @@ func (h Handler) onChange(key string, obj *webhookv1.GitCommit) (*webhookv1.GitC
...
+	if len(obj.Status.Endpoints) > 0 {
+		endpoint = obj.Status.Endpoints[0]
+	}
+	gitcommit.Status.GithubStatus.LogURL = logURL
+	gitcommit.Status.GithubStatus.EnvironmentURL = endpoint
+	state := "in_progress"
+	if serviceset.IsReady(obj.Status.DeploymentStatus) {
+		state = "success"
+	}
+	gitcommit.Status.GithubStatus.DeploymentState = state
```
Comment (ibuildthecloud) [inline modules/build/controllers/gitcommit/controller.go]: Don't update unless changed

File: modules/build/controllers/service/service.go [hunk 3]
```diff
@@ -101,6 +104,15 @@ func (p *populator) populate(obj runtime.Object, ns *corev1.Namespace, os *objec
 		return nil
 	}
 
+	if service.Status.BuildLogToken == "" {
+		service.Status.BuildLogToken = uuid.New().String()
```
Comment (ibuildthecloud) [inline modules/build/controllers/service/service.go]: copy rancher/pkg/randomtoken

File: pkg/apis/rio.cattle.io/v1/service_types.go [hunk 4]
```diff
@@ -399,4 +409,7 @@ type ImageBuild struct {
 
 	// Push image
 	Push bool `json:"push,omitempty"`
+
+	// Disable Github Deployment status update
+	DisableGithubDeployment bool `json:"disableGithubDeployment,omitempty"`
```
Comment (ibuildthecloud) [inline pkg/apis/rio.cattle.io/v1/service_types.go]: Drop this option

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rio/issues/406

**Pull request merged**